pytorch_lightning/metrics/metric.py

```diff
@@ -159,38 +197,17 @@ class TensorMetric(Metric):
     Already handles DDP sync and input/output conversions.
     """
 
-    def __init__(self, name: str,
-                 reduce_group: Optional[Any] = None,
-                 reduce_op: Optional[Any] = None):
-        """
-
-        Args:
-            name: the metric's name
-            reduce_group: the process group for DDP reduces (only needed for DDP training).
-                Defaults to all processes (world)
-            reduce_op: the operation to perform during reduction within DDP (only needed for DDP training).
-                Defaults to sum.
-        """
-        super().__init__(name)
-        self.reduce_group = reduce_group
-        self.reduce_op = reduce_op
-
     @staticmethod
     def input_convert(self, data: Any):
-        return apply_to_collection(data,
-                                   (torch.Tensor, np.ndarray, numbers.Number),
-                                   convert_to_tensor,
-                                   self.dtype, self.device)
+        data = apply_to_collection(
+            data, (torch.Tensor, np.ndarray, numbers.Number), convert_to_tensor, self.dtype, self.device
+        )
+        return super(TensorMetric, self).input_convert(self, data)
```

Comment:
Is it important to name the class in the super call?

Reply:
It is, since this is a static method and thus the super call cannot infer the base class otherwise.

Unfortunately we also can't make this a non-static method since PyTorch requires all it's hooks to get the module passed explicitly

Reply:
got it!